# HW1 Part 2 - MGT 6203

## Instructions:

**For the rest of the semester, we will be using Vocareum to work on and submit your homework assignments. Vocareum is a cloud platform for programming classes. It provides an infrastructure that allows us to move the educational aspects like assignments, exams, quizzes, etc, to the cloud. The merit of this platform is that all of you will be working in the same coding environment. This way we can eliminate many issues we might encounter when working on an individual basis, such as those with library installations and RStudio malfunctions. Some you you might be new to this platform, here we provided a few things to keep in mind to get you started, please try to read through them.**

**Things to keep in mind:**


<ol>
<li>Even though we are moving from your local envrionment to the cloud, **our expectations from your homework will stay the same**. Same goes for the rubrics.</li>

<li>Vocareum has its own cloud based file system, the data files you will be using for the assignments will be stored in the cloud with path **"../resource/asnlib/publicdata/FILENAME.csv"**. You will be able to import them with the same method as you do in RStudio, simply substitute the path name to the one specified in the instructions. You won't be able to modify these data files.</li>

<li>You will be able to find the data files on Canvas/EdX if you would like to explore them offline.</li>

<li>**For coding questions, you will be graded on the R code as well as the output** in your submission.</li>

<li>**For interpretations or short response questions, please type the answers in the notebook's markdown cells**. To change a code cell to a markdown cell, click on the cell, and in the dropdown menu above, switch the type of the cell block from "code" to "markdown". **Adding print statements to code cells for short response/interpretation questions is also fine, as long as we can clearly see the output of your response**.</li>

<li>You don't need to, but if you would like to learn more about how to format your markdown cells, visit the following site: https://www.earthdatascience.org/courses/intro-to-earth-data-science/file-formats/use-text-files/format-text-with-markdown-jupyter-notebook/. Jupyter notebook also support LaTeX.</li>

<li>**Feel free to add as many additional cells as you need**. But please keep your solution to a question directly under that question to aviod confusions.</li>

<li>You may delete the #SOLUTION BEGINS/ENDS HERE comments from the cell blocks, they are just pointers that indicates where to put you solutions.</li>

<li>When you have finished the assignment, remember to rerun your notebook to check if it runs correctly. You can do so by going to **Kernel-> Restart & Run All**. You may lose points if your solutions does not run successfully.</li>

<li>**Click the "Submit" button on the top right corner to turn in your assignment**. Your assignment will enter the next phase for peer review.</li>

<li>You will be able to come back and resubmit your assignment as long as it is before the start of the peer review period.</li>

</ol>



# About Package Installation:

Most of the packages (if not all) that you will need to complete this assignment are already installed in this environment. An easy way to check this is to run the command: library(PackageName). If this command runs successfully then the package was already installed and has been successfully attached to the code. If the command gave an error saying the Package was not found then follow the steps below to successfully install the package and attach it to the code: 

Use *installed.packages()* command to return a table of the packages that are preinstalled in the environment.

To attach a preinstalled library in Vocareum, simply use *library(PackageName)*

**To install a package that does not come with the provided environment, please use the following syntax:**

*install.packages("PackageName", lib="../work/")*

**To attach a library you just installed, use the following syntax:**

*library(PackageName, lib.loc="../work/")*

Make sure the file location is the same as the above code snippets *("../work/")*

## Q1. Use the *"airbnb_data.csv"* provided and answer the following questions on Linear Regression:

**Instruction: The file "airbnb_data.csv" can be accessed at the path: ("../resource/asnlib/publicdata/airbnb_data.csv")**

a)  Remove ‘id’columns (‘room_id’, ‘survey_id’, ‘host_id’) and ‘city’ from your dataset, and fit a multiple linear regression model using price as the response variable and all others as predictor variables.  (Note: Do not fit a model using id columns and city as predictors). Which variables are statistically significant at a 95% confidence interval.    (6 points)

In [1]:
# SOLUTION BEGINS HERE
library(ggplot2) 
library(tidyverse) 

# read the csv data and initialize column types
airbnb_data = read_csv("../resource/asnlib/publicdata/airbnb_data.csv", col_types = list(
    room_type = col_factor(c('Shared room','Private room','Entire home/apt')),
    reviews = col_integer(),
    overall_satisfaction = col_double(),
    accommodates = col_integer(),
    bedrooms = col_integer(),
    price = col_integer()))

# remove 'id' columns and 'city' column
keep = c(4,6,7,8,9,10)
airbnb_data = airbnb_data[keep]
attach(airbnb_data)

# fir a multiple linear regression model using the dataset above
RS.lm <- lm(price ~., data = airbnb_data)

# summary info of the model
summary(RS.lm)

# answer in markdown below
# SOLUTION ENDS HERE

── Attaching packages ─────────────────────────────────────── tidyverse 1.2.1 ──
✔ tibble  2.1.3     ✔ purrr   0.2.5
✔ tidyr   1.0.0     ✔ dplyr   0.8.3
✔ readr   1.1.1     ✔ stringr 1.3.1
✔ tibble  2.1.3     ✔ forcats 0.3.0
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()



Call:
lm(formula = price ~ ., data = airbnb_data)

Residuals:
   Min     1Q Median     3Q    Max 
-367.8  -49.2    3.2   38.6 4032.7 

Coefficients:
                           Estimate Std. Error t value Pr(>|t|)    
(Intercept)              -100.02951   62.59071  -1.598  0.11038    
room_typePrivate room      75.73664   60.60664   1.250  0.21178    
room_typeEntire home/apt   76.66780   59.90939   1.280  0.20099    
reviews                     0.01090    0.09982   0.109  0.91310    
overall_satisfaction      -10.48160    3.47320  -3.018  0.00262 ** 
accommodates               23.00721    5.23952   4.391 1.27e-05 ***
bedrooms                   85.64533   11.45983   7.474 1.95e-13 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 167.1 on 847 degrees of freedom
Multiple R-squared:  0.3228,	Adjusted R-squared:  0.318 
F-statistic:  67.3 on 6 and 847 DF,  p-value: < 2.2e-16


In [2]:
contrasts(room_type)

Private room Entire home/apt
Shared room     0            0              
Private room    1            0              
Entire home/apt 0            1

> **ANSWER TO QUESTION 1a:**

> Overall_satisfaction, accommodates, and bedrooms are statistically significant variables at a 95% confidence interval as their p(s) are < 0.05

In [3]:
head(airbnb_data)

room_type   reviews overall_satisfaction accommodates bedrooms price
1 Shared room   0     0.0                  4            1        67   
2 Shared room  32     5.0                  4            1        76   
3 Shared room   4     4.5                  2            1        45   
4 Shared room  24     4.5                  6            1        26   
5 Shared room 152     4.5                  6            1        26   
6 Shared room  20     4.5                  4            1        26

b) Interpret the coefficients for predictors: 'room_type', 'bedrooms'. (3 points)

**Note:** In the event that a coefficient is not significant in your analysis, please interpret both coefficients as if they were significant. 

In [4]:
# SOLUTION BEGINS HERE
# Answer in the Markdown below
# SOLUTION ENDS HERE

> **ANSWER TO QUESTION 1b:**

> As we have:
> Price = b<sub>0</sub> + b<sub>1</sub>Entire home/apt + b<sub>2</sub>Private room + ... + b<sub>6</sub>bedrooms 

> The baseline is **Shared room** as indicated from running function constracts() above:

> For every one unit changed in **Entire home/apt** room type, the price is expected to increase 76.66780 keeping all esle constant.

> For every one unit changed in **Private room** room type, the price is expected to increase 75.73664 keeping all esle constant.

> For every one unit changed in **bedrooms**, the price is expected to increase 85.64533 keeping all esle constant.

c) Predict the price (nearest dollar) for a listing with the following factors: 'bedrooms' = 1, 'accommodates' = 2, 'reviews' = 70, 'overall_satisfaction' = 4, and 'room_type' = 'Private room'. (6 points)



|bedrooms|accomodates|reviews|overall_satisfaction|room_type|
|------|------|------|-------|------|------|
|1|2|70|4|Private room|


In [5]:
# SOLUTION BEGINS HERE
new_data <- data.frame(bedrooms = 1, accommodates = 2, reviews = 70, overall_satisfaction = 4, room_type = "Private room")
predict(RS.lm, new_data, interval="predict")
# answer to question 1c in the markdown below
# SOLUTION ENDS HERE

fit      lwr       upr     
1 66.20316 -262.3605 394.7669

> **ANSWER TO QUESTION 1c:**
> From the calculation above, the predicted **Price** is 66.20


d) Identify outliers using Cook's distance approach. Remove points having Cook's distance > 1. Rerun the model after the removal of these points and print the summary. (6 points)

**Note:** Please build the same model structure as used on parts 1a-1b, but using the new dataset with outliers removed.

In [6]:
# SOLUTION BEGINS HERE
cooksD <- cooks.distance(RS.lm)
influential <- cooksD[(cooksD > 1)]
influential
airbnb_without_outliers <- airbnb_data[c(-94,-95),]
RS2.lm <- lm(price ~., data = airbnb_without_outliers)
summary(RS2.lm)
# SOLUTION ENDS HERE

94       95 
1.117826 6.846108


Call:
lm(formula = price ~ ., data = airbnb_without_outliers)

Residuals:
    Min      1Q  Median      3Q     Max 
-190.95  -32.43   -7.09   20.35  876.26 

Coefficients:
                          Estimate Std. Error t value Pr(>|t|)    
(Intercept)              -16.68641   25.40747  -0.657 0.511519    
room_typePrivate room     59.41750   24.56885   2.418 0.015799 *  
room_typeEntire home/apt  91.69951   24.28958   3.775 0.000171 ***
reviews                   -0.05915    0.04047  -1.462 0.144202    
overall_satisfaction      -6.78957    1.41118  -4.811 1.78e-06 ***
accommodates              11.90698    2.14267   5.557 3.68e-08 ***
bedrooms                  35.93177    4.87968   7.364 4.25e-13 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 67.73 on 845 degrees of freedom
Multiple R-squared:  0.4249,	Adjusted R-squared:  0.4208 
F-statistic:   104 on 6 and 845 DF,  p-value: < 2.2e-16


e) Using the original dataset (not the one created in 1D) create another multiple linear regression model by removing the 'id' columns ('room_id', survery_id', 'host_id'), 'city', and 'room_type' from the Airbnb_data.csv dataset, run the vif function and calculate the vif value of the variable 'bedrooms' (3 points)

In [7]:
# SOLUTION BEGINS HERE
library(car)

# remove room_type from the dataset
new_airbnb_data = airbnb_data[-1]

# running the multiple linear regression model
RS3.lm <- lm(price ~., data = new_airbnb_data)

# run the vif function and calculate vif values
vif(RS3.lm)
# answer to 2e in the markdown below
# SOLUTION ENDS HERE

Loading required package: carData

Attaching package: ‘car’

The following object is masked from ‘package:dplyr’:

    recode

The following object is masked from ‘package:purrr’:

    some



reviews overall_satisfaction         accommodates 
            1.136063             1.129636             2.761613 
            bedrooms 
            2.759746

> VIF values of the variable bedrooms is 2.7597457287341

## Q2. Use the *"direct_marketing.csv"* provided and answer the following questions on Linear Regression:

**Create indicator variables for the 'History' column. Considering the base case as None (i.e., create Low, Medium and High variables with 1 denoting the positive case and 0 the negative) and few additional variables LowSalary, MediumSalary and HighSalary based on the customer history type i.e., MediumSalary = Medium*Salary etc.**

**Instruction: The dataset "direct_marketing.csv" can be accessed at the path: ("../resource/asnlib/publicdata/direct_marketing.csv")**


a) Fit a multiple linear regression model using 'AmountSpent' as the response variable and the indicator variables along with their salary variables as the predictors: (6 points)

$$AmountSpent = \beta_0 + \beta_1Salary + \beta_2Low + \beta_3Medium + \beta_4High + \beta_5LowSalary + \beta_6MediumSalary + \beta_7HighSalary$$

In [8]:
# SOLUTION BEGINS HERE
data <- read_csv("../resource/asnlib/publicdata/direct_marketing.csv", col_types = list(
    Age = col_character(),
    Gender = col_character(),
    OwnHome = col_character(),
    Married = col_character(),
    Location = col_character(),
    Salary = col_integer(),
    Children = col_integer(),
    History = col_factor(c("None", "Low", "Medium", "High")),
    Catalogs = col_integer(),
    AmountSpent = col_double()))
attach(data)
                
Model = lm(AmountSpent ~ Salary + History + Salary*History, data=data)
summary(Model)
# SOLUTION ENDS HERE


Call:
lm(formula = AmountSpent ~ Salary + History + Salary * History, 
    data = data)

Residuals:
    Min      1Q  Median      3Q     Max 
-214.33  -25.47   -6.46   20.64  352.50 

Coefficients:
                       Estimate Std. Error t value Pr(>|t|)    
(Intercept)           1.9622199  6.3880253   0.307 0.758777    
Salary                0.0023641  0.0001071  22.083  < 2e-16 ***
HistoryLow           25.4466733  8.9203292   2.853 0.004426 ** 
HistoryMedium        79.2984388 12.8982169   6.148 1.14e-09 ***
HistoryHigh          72.6735221 15.2270169   4.773 2.09e-06 ***
Salary:HistoryLow    -0.0021069  0.0001890 -11.150  < 2e-16 ***
Salary:HistoryMedium -0.0021153  0.0002182  -9.693  < 2e-16 ***
Salary:HistoryHigh   -0.0006408  0.0001926  -3.328 0.000908 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 55.79 on 992 degrees of freedom
Multiple R-squared:  0.6654,	Adjusted R-squared:  0.6631 
F-statistic: 281.9 on 7 and 992 DF,  p-valu

b) What is the amount spent by a customer for each historic type (None, Low, Medium, and High) provided their salary is $10,000 based on the model constructed in part a? (6 points)
    
    
    
    
    

In [9]:
# SOLUTION BEGINS HERE
print("None type:")
predict(Model, data.frame(History='None', Salary = 10000))
print("Low type:")
predict(Model, data.frame(History='Low', Salary = 10000))
print("Medium type:")
predict(Model, data.frame(History='Medium', Salary = 10000))
print("High type:")
predict(Model, data.frame(History='High', Salary = 10000))
# SOLUTION ENDS HERE

[1] "None type:"


1 
25.60347

[1] "Low type:"


1 
29.98157

[1] "Medium type:"


1 
83.74909

[1] "High type:"


1 
91.86874

## Q3. Use the *"airbnb_data.csv"* provided and answer the following questions (part c and part d) on Linear Regression.  DO NOT remove outliers from the dataset:  Perform Log transformation for the variables price and overall_satisfaction, make necessary transformations suggested in the class.

**Instruction: The dataset "airbnb_data.csv" can be accessed at the path: ("../resource/asnlib/publicdata/airbnb_data.csv")**

a) Fit all four models i.e., linear-linear, linear-log, log-linear and log-log regression models using price as the response variable and overall_satisfaction as the predictor.  (Note: Because overall_satisfaction contains ‘0’ values, you will need to use log(x+1) transformations instead of log(x) transformations) (6 points)

In [10]:
# SOLUTION BEGINS HERE
airbnb_data = read_csv("../resource/asnlib/publicdata/airbnb_data.csv")
# keep = c(4,6,7,8,9,10)
# airbnb_data = airbnb_data[keep]
attach(airbnb_data)
overall_satisfaction_log <- log(overall_satisfaction + 1)
linear_linear <- lm(price ~ overall_satisfaction, data = airbnb_data)
linear_log <- lm(price ~ overall_satisfaction_log, data = airbnb_data)
log_linear <- lm(log(price) ~ overall_satisfaction, data = airbnb_data)
log_log <- lm(log(price) ~ overall_satisfaction_log, data = airbnb_data)
# SOLUTION ENDS HERE

Parsed with column specification:
cols(
  room_id = col_integer(),
  survey_id = col_integer(),
  host_id = col_integer(),
  room_type = col_character(),
  city = col_character(),
  reviews = col_integer(),
  overall_satisfaction = col_double(),
  accommodates = col_integer(),
  bedrooms = col_integer(),
  price = col_integer()
)
The following objects are masked from airbnb_data (pos = 6):

    accommodates, bedrooms, overall_satisfaction, price, reviews,
    room_type



b) Which of the four models has the best  R2R2 ?

Discuss how these transformations impact model fit. Does log transforming the variable improve, degrade, or have an uncertain impact on model performance?

Compare the models and their performance with respect to R squared. Are these R squared values generally good, bad, or uncertain across the four models as a general measure of fit? (2 points)

In [11]:
# SOLUTION BEGINS HERE
print("linear-linear model R Squared:")
summary(linear_linear)$r.squared
print("linear-log model R Squared:")
summary(linear_log)$r.squared
print("log-linear model R Squared:")
summary(log_linear)$r.squared
print("log-log model R Squared:")
summary(log_log)$r.squared
# Answer to question 3b is the markdown below
# SOLUTION ENDS HERE

[1] "linear-linear model R Squared:"


[1] 0.02018428

[1] "linear-log model R Squared:"


[1] 0.02088739

[1] "log-linear model R Squared:"


[1] 0.01777027

[1] "log-log model R Squared:"


[1] 0.01933861

> **ANSWER TO QUESTION 3b:**
> From the result of R<sup>2</sup> above, we can tell that "linear-log" model has the highest R<sup>2</sup> result. However, since all the R<sup>2</sup> value of all 4 models are too low, we can say that the independent variable (overall satisfaction) is not explaining much in the variation of dependent variable (price). R<sup>2</sup> values generally uncertain across the four models as a general measure of fit.


## Q4. The attached "*titanic_data.csv"* file was obtained from the following source: http://math.ucdenver.edu/RTutorial/

**It has been cleaned to remove all rows which contain missing values. We will perform a logistic regression on this cleaned dataset.**

**The dataset contains the following columns:**


|Column Name|Description|Data Type|
|------|------|------|
|Name |Passenger Name|factor|
|PClass|Passenger Class (1st, 2nd, 3rd)|factor|
|Age|Passenger Age|number|
|Sex|Passenger Sex - female, male|factor|
|Survived|1 if passenger survived, 0 if not|number|

**After converting the survived variable to be a factor with two levels, 0 and 1, perform a logistic regression on the dataset using 'survived' as the response and 'Sex' as the explanatory variable.**

**Instruction: The file "titanic_data.csv" can be accessed at the path: ("../resource/asnlib/publicdata/titanic_data.csv")**

a) Display the model summary. What does the value of the intecept coefficient represent in this model? (3 points)


In [12]:
# SOLUTION BEGINS HERE
titanic_data <- read_csv("../resource/asnlib/publicdata/titanic_data.csv")
attach(titanic_data)
titanic_data$Survived <- as.factor(titanic_data$Survived)
titanic_data$Sex <- as.factor(titanic_data$Sex)

Model.lm = glm(Survived ~ Sex, data = titanic_data, family = "binomial")
summary(Model.lm)
# SOLUTION ENDS HERE

Parsed with column specification:
cols(
  Name = col_character(),
  PClass = col_character(),
  Age = col_double(),
  Sex = col_character(),
  Survived = col_integer()
)
The following object is masked from data:

    Age




Call:
glm(formula = Survived ~ Sex, family = "binomial", data = titanic_data)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-1.6735  -0.6776  -0.6776   0.7524   1.7800  

Coefficients:
            Estimate Std. Error z value Pr(>|z|)    
(Intercept)   1.1172     0.1367   8.171 3.05e-16 ***
Sexmale      -2.4718     0.1783 -13.861  < 2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 1025.57  on 755  degrees of freedom
Residual deviance:  796.64  on 754  degrees of freedom
AIC: 800.64

Number of Fisher Scoring iterations: 4


In [20]:
head(titanic_data)

Name                                          PClass Age   Sex    Survived
1 Allen, Miss Elisabeth Walton                  1st    29.00 female 1       
2 Allison, Miss Helen Loraine                   1st     2.00 female 0       
3 Allison, Mr Hudson Joshua Creighton           1st    30.00 male   0       
4 Allison, Mrs Hudson JC (Bessie Waldo Daniels) 1st    25.00 female 0       
5 Allison, Master Hudson Trevor                 1st     0.92 male   1       
6 Anderson, Mr Harry                            1st    47.00 male   1       
  first_class second_class
1 1           0           
2 1           0           
3 1           0           
4 1           0           
5 1           0           
6 1           0

> **ANSWER TO QUESTION 4a:**
> The value of the intecept coefficient represents the log odds of being survived for a female

b) Determine the probability of survival for females. (3 points)



In [13]:
Model.lm$coefficients[1]

(Intercept) 
   1.117217

In [14]:
# SOLUTION BEGINS HERE
# The intercept b0 = 1.1172 is the log odds for female (base case)
odd <- exp(Model.lm$coefficients[1])
print(paste("The odd is: ", odd))
p <- odd/(1+odd)
print(paste("The probability of survival for female is: ", p))
# SOLUTION ENDS HERE

[1] "The odd is:  3.05633802816021"
[1] "The probability of survival for female is:  0.753472222221687"


c) Determine the probability of survival for males. (3 points)


In [15]:
Model.lm$coefficients[2]

Sexmale 
-2.471763

In [16]:
# SOLUTION BEGINS HERE
odd <- exp(Model.lm$coefficients[1] + Model.lm$coefficients[2])
print(paste("The odd is: ", odd))
p1 <- odd/(1+odd)
print(paste("The probability of survival for male is: ", p1))
# SOLUTION ENDS HERE

[1] "The odd is:  0.258064516129129"
[1] "The probability of survival for male is:  0.205128205128266"


d) Create indicator variables for the variable 'PClass'. Consider PClass='3rd' as the base case (first_class = 1 if PClass is 1st or else 0; second_class = 1 if PClass is 2nd or else 0). 

Create a logistic regression model with the variable 'Survived' as the response variable and the indicator variables created for 'PClass' as the independent model variables. 

Using this model, calculate the probability of survival for the passengers in all 3 classes. (7 points)

In [17]:
head(titanic_data)

Name                                          PClass Age   Sex    Survived
1 Allen, Miss Elisabeth Walton                  1st    29.00 female 1       
2 Allison, Miss Helen Loraine                   1st     2.00 female 0       
3 Allison, Mr Hudson Joshua Creighton           1st    30.00 male   0       
4 Allison, Mrs Hudson JC (Bessie Waldo Daniels) 1st    25.00 female 0       
5 Allison, Master Hudson Trevor                 1st     0.92 male   1       
6 Anderson, Mr Harry                            1st    47.00 male   1

In [18]:
# SOLUTION BEGINS HERE
# create indicator variables for the variable "PClass"
titanic_data$first_class <- ifelse(titanic_data$PClass == '1st',1,0)
titanic_data$second_class <- ifelse(titanic_data$PClass == '2nd',1,0)

# run logistic regression model
new_md = glm(Survived ~ first_class + second_class, data = titanic_data, family = "binomial")
summary(new_md)

# Calculate the probability of survival for the passengers in all 3 classes
odd <- exp(new_md$coefficients[1])
print(paste("The odd is: ", odd))
p <- odd/(1+odd)
print(paste("The probability of survival for 3rd class is: ", p))
odd <- exp(new_md$coefficients[1] + new_md$coefficients[2])
print(paste("The odd is: ", odd))
p <- odd/(1+odd)
print(paste("The probability of survival for 1st class is: ", p))
odd <- exp(new_md$coefficients[1] + new_md$coefficients[3])
print(paste("The odd is: ", odd))
p <- odd/(1+odd)
print(paste("The probability of survival for 2nd class is: ", p))
# SOLUTION ENDS HERE


Call:
glm(formula = Survived ~ first_class + second_class, family = "binomial", 
    data = titanic_data)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-1.3818  -1.0982  -0.7502   0.9860   1.6765  

Coefficients:
             Estimate Std. Error z value Pr(>|z|)    
(Intercept)   -1.1239     0.1303  -8.623  < 2e-16 ***
first_class    1.5925     0.1889   8.431  < 2e-16 ***
second_class   0.9347     0.1898   4.925 8.45e-07 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 1025.57  on 755  degrees of freedom
Residual deviance:  947.55  on 753  degrees of freedom
AIC: 953.55

Number of Fisher Scoring iterations: 4


[1] "The odd is:  0.325000000000923"
[1] "The probability of survival for 3rd class is:  0.24528301886845"
[1] "The odd is:  1.59770114942528"
[1] "The probability of survival for 1st class is:  0.61504424778761"
[1] "The odd is:  0.827586206896552"
[1] "The probability of survival for 2nd class is:  0.452830188679245"


In [19]:
new_md$coefficients[3]

second_class 
   0.9346881